web/files/files.go

```diff
@@ -76,5 +184,65 @@ func Upload(c *gin.Context) {
 
 // Routes sets the routing for the files service
 func Routes(router *gin.RouterGroup) {
-	router.POST("/:folder-id", Upload)
+	router.POST("/", CreationHandler)
+	router.POST("/:folder-id", CreationHandler)
+}
+
+func makeCode(err error) (code int) {
+	switch err {
+	case errDocAlreadyExists:
+		code = http.StatusConflict
+	default:
+		code = http.StatusInternalServerError
+	}
+	return
+}
+
+func parseTags(str string) []string {
+	var tags []string
+	for _, tag := range strings.Split(str, ",") {
+		// @TODO: more sanitization maybe ?
```

Comment:
I think you can remove this TODO. I don't see a reason to restrict what the tags can be